src/components/sections/contest/ContestDetails/ContestInfo.tsx

```diff
@@ -69,6 +90,77 @@ const ContestInfo = ({
         <h3 className="text-bodyL text-purple-light">Custom conditions</h3>
         <p className="text-bodyM">{contest.customConditions || '-'}</p>
       </div>
+
+      {session?.user.role === UserRole.JUDGE &&
+        contest.status === ContestStatus.IN_REVIEW && (
+          <div className="flex gap-8">
+            <AlertDialog>
+              <AlertDialogTrigger>
+                <Button variant="outline" className="flex gap-3">
+                  <span className="uppercase">Approve contest</span>
+                </Button>
+              </AlertDialogTrigger>
+
+              <AlertDialogContent>
+                <AlertDialogHeader>
+                  <AlertDialogTitle className="uppercase">
+                    Are you sure you want to approve this contest?
+                  </AlertDialogTitle>
+                </AlertDialogHeader>
+                <AlertDialogFooter>
+                  <AlertDialogCancel>Cancel</AlertDialogCancel>
+                  <AlertDialogAction
+                    onClick={() =>
+                      mutate(
+                        {
+                          contestId: contest.id,
+                          newStatus: ContestStatus.APPROVED,
+                        },
+                        {
+                          onSuccess: () => router.push(PATHS.judgeContests),
+                        },
+                      )
+                    }>
```

Comment:
I don't really like functions inside JSX, that are longer than a couple of lines. Consider extracting this into some `approveContest` functions instead. Same for the `reject` function.